### ISSUE TITLE
Implementation of Difficulty-Aware Retrospective Sampling

### ISSUE THREAD

**Issue by hjh0119** [Tongyi-Zhiwen/Qwen-Doc#1]
Thank you for your work. I checked the codebase, but I couldn't find the implementation related to **Difficulty-Aware Retrospective Sampling** in the paper. Where can I find that part of the code?

**Comment by fanqiwan** [comment]
Hello. Since this is an offline data selection strategy to filter difficult instances for RL training, we do not add this in the training code for simplification.

To implement this strategy, we just need to generate multiple responses for a given question with vLLM, then calculate reward for each response, finally select the difficult questions with low rewards.
